lib/pbench/server/auth/__init__.py

```diff
@@ -31,11 +32,163 @@ class OpenIDCAuthenticationError(OpenIDClientError):
     pass
 
 
+class OpenIDTokenInvalid(Exception):
+    pass
+
+
+class Connection:
+    """Helper connection class for use by an OpenIDClient instance."""
+
+    def __init__(
+        self,
+        server_url: str,
+        headers: Optional[Dict[str, str]] = None,
+        verify: bool = True,
+    ):
+        self.server_url = server_url
+        self.headers = CaseInsensitiveDict({} if headers is None else headers)
+        self.verify = verify
+        self._connection = requests.Session()
+
+    def _method(
+        self,
+        method: str,
+        path: str,
+        data: Union[Dict, None],
+        headers: Optional[Dict] = None,
+        **kwargs,
+    ) -> requests.Response:
+        """Common frontend for the HTTP operations on OIDC client connection.
+
+        Args:
+            method : The API HTTP method
+            path : Path for the request.
+            data : Json data to send with the request in case of the POST
+            kwargs : Additional keyword args
+
+        Returns:
+            Response from the request.
+        """
+        final_headers = self.headers.copy()
+        if headers is not None:
+            final_headers.update(headers)
+        try:
+            response = self._connection.request(
+                method,
+                urljoin(self.server_url, path),
+                params=kwargs,
+                data=data,
+                headers=final_headers,
+                verify=self.verify,
+            )
+        except (
+            requests.exceptions.ConnectionError,
+            requests.exceptions.Timeout,
+        ) as exc:
+            raise OpenIDClientError(
+                HTTPStatus.BAD_GATEWAY,
+                f"Failure to connect to the OpenID client, {self!r}: {exc}",
+            )
```

Comment:
`OpenIDCClientError` (and others) should probably extend `APIAbort` so they're handled cleanly in `_dispatch` instead of going to the catch-all.

Also, where you use `HTTPStatus.INTERNAL_SERVER_ERROR` it'd be better to use `APIInternalError` so we generate the logger uuid right away rather than, again, going through the `Exception` catch-all in `_dispatch`.